### PR TITLE
Added pause & reset function. Also added a little help text.

### PR DESCRIPTION
**Pull request by JannesPeters**
@lennet hope this is the right way to do it. Also not sure how 'good' the code is, but it works for me without ever crashing 🤞

### CODE REVIEW

File: code.ts [hunk 1]
```diff
@@ -1,10 +1,49 @@
 const delay = ms => new Promise(res => setTimeout(res, ms));
 var activeTimer = 0;
 const secondsSet = [86400, 3600, 60, 1];
+const progressBar = figma.createRectangle();
+var pause = 0;
+var reset = 0;
```
Comment (lennet) [inline code.ts:6]: Is  there a reason for not using a boolean here?

File: code.ts [hunk 2]
```diff
@@ -1,10 +1,49 @@
 const delay = ms => new Promise(res => setTimeout(res, ms));
 var activeTimer = 0;
 const secondsSet = [86400, 3600, 60, 1];
+const progressBar = figma.createRectangle();
```
Comment (lennet) [inline code.ts:4]: It looks like this variable is never used so we could delete that line
```suggestion
```

File: code.ts [hunk 3]
```diff
@@ -1,10 +1,49 @@
...
+    pause = 1;
+  }
+
+  if (msg.type === 'helpon'){
+    figma.ui.resize(220,150);
+  }
+
+  if (msg.type === 'helpoff'){
+    figma.ui.resize(220,50);
+  }
```
Comment (lennet) [inline code.ts:38]: This could be optimised with a switch-case.  Similar to the logic in ui.html

File: code.ts [hunk 4]
```diff
@@ -98,20 +137,61 @@ async function startTimer(node: TextNode, seconds: number, template: string, sta
   await figma.loadFontAsync(node.fontName as FontName);
   activeTimer += 1;
 
+  console.log("Timer started / became active");
+  
+  var timerID = activeTimer;
+  var keepItRunning = 1;
```
Comment (lennet) [inline code.ts:143]: same here. Let's use a boolean

File: code.ts [hunk 5]
```diff
@@ -1,10 +1,49 @@
...
+    figma.ui.resize(220,150);
+  }
+
+  if (msg.type === 'helpoff'){
+    figma.ui.resize(220,50);
+  }
+};
+
+function checkAndStart()
+{
```
Comment (lennet) [inline code.ts:42]: small formatting improvement:
```suggestion
function checkAndStart() {
```

File: code.ts [hunk 6]
```diff
@@ -98,20 +137,61 @@ async function startTimer(node: TextNode, seconds: number, template: string, sta
...
 
+  console.log("Timer started / became active");
+  
+  var timerID = activeTimer;
+  var keepItRunning = 1;
   var secondsToGo = seconds;
-  while(secondsToGo > 0) {
-    var newText = fillUpTimeStringWithTempalte(secondsToInterval(secondsToGo), template);
-    if (startsWithTimer) {
+  var newText = "not set";
```
Comment (lennet) [inline code.ts:145]:
```suggestion
  var newText = "";
```

**Comment by lennet** [comment]
Thanks for your contribution, I like the new feature!

I left you some comments. Overall it would be nice if you could:

- [ ] Remove dead code (logs, unreachable code or code that is commented out)
- [ ] Use boolean instead of integers (e.g. `var paused = false` instead of `var paused = 0` )
- [ ] Apply a code formatter to the files you changed (in VS Code you can right click and select Format Document)

File: code.ts (hunk 1 above)
Comment (JannesPeters) [reply code.ts:5]: just changed it to boolean

File: code.ts (hunk 3 above)
Comment (JannesPeters) [reply code.ts:37]: just implemented the switch-case

File: code.ts (hunk 4 above)
Comment (JannesPeters) [reply code.ts:141]: just changed to boolean

File: code.ts [hunk 7]
```diff
@@ -98,20 +143,42 @@ async function startTimer(node: TextNode, seconds: number, template: string, sta
...
+        }
+        node.characters = newText;
+        secondsToGo -= 1;
+      } else if (secondsToGo < 1) {
+        node.characters = "Done";
+      }
+      await delay(1000);
+    } else {
+      await delay(1000);
     }
```
Comment (lennet) [inline code.ts:179]: Looks like we don't need the else case here and we can add the delay after the if block

File: code.ts [hunk 8]
```diff
@@ -98,20 +143,42 @@ async function startTimer(node: TextNode, seconds: number, template: string, sta
...
-        newText = "Timer: " + newText;
+  var newText = "";
+
+  while (keepItRunning) {
+
+    // checking if reset was clicked by user and if so resetting all timers
+    if (reset) {
+      secondsToGo = seconds;
+      newText = fillUpTimeStringWithTemplate(secondsToInterval(secondsToGo), template);
+      newText = "Timer: " + newText;
```
Comment (lennet) [inline code.ts:159]: Not all Timer start with "Timer: "

Bug

File: ui.html [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    background: lightgray;
+  }
+</style>
+
+<button id="start" style="width: 70;">Start</button>
+<button id="reset">Reset</button>
+<button id="help">?</button>
+<b id="helptext" hidden> <br /> <br /> Type the time you want to count down in "Timer: HH:MM:SS" format in any text
+  field (e.g. Timer: 5:00) then press 'start' </b>
+
```
Comment (lennet) [inline ui.html:30]: Maybe we should also add a hint that selecting that doesn't start with "Timer: " also works

**Pull request merged**